### ISSUE TITLE
[KT2021] Include faction-specific strategy ploys and tactical ploys ?

### ISSUE THREAD

**Issue by lstrzebinczyk** [Floppy/dataslate#547]
What do you think about that idea? Looks like the roster file has a string representing which faction that is, so faction-specific strategy and tactical ploys could be automatically added. 
On a similiar fashion, faction items (the ones you buy for EP) could be added.

I know I would use that, and I can also prepare a PR with initial version : )

**Comment by Floppy** [comment]
And Tac Ops too.

But, this is a tricky one. It would *definitely* be useful, I won't deny that. However, as currently *all* the data comes from the Battlescribe rosters and data files, there's a couple of tricky aspects.

1) We could get the BSData maintainers to add them in, like they have the Guardsman orders reference. This has the advantage that we don't add the data here, but I know that they've resisted stuff like that in the past, as it's not supposed to be a complete rules references. @Mad-Spy, the KT maintainer for BSData, will have thoughts here.

2) If we included them in this app, I'd be scared of opening us up to copyright strikes from GW, although they've not done that to Battlescribe, so it might not be a concern?

Basically, if we can decide the right way to do this and be happy that it won't get us in trouble, I'd love to have it here.

**Comment by lstrzebinczyk** [comment]
I was thinking about including them in this repository. If you are ok with it, I can get some work done, but that's your decision to make.

**Comment by Mad-Spy** [comment]
The Battlescribe data won't include them; it's a list building aid, not a playing aid. Also, the output format of BS isn't great for anything like that - hence the need for this app in the first place!

**Comment by Floppy** [comment]
OK, @lstrzebinczyk if you want to have a go, go for it. I'd like it if we could keep the data as separate as possible, so design with that in mind :)

**Comment by lstrzebinczyk** [comment]
I will give it a go over the weekend!

**Comment by Floppy** [comment]
I'm going to have a crack at the grey knights psychic power and guard order reference display, so there might be some useful stuff in that for you to reuse.

**Comment by lstrzebinczyk** [comment]
@Floppy Sorry I couldn't open a PR over the weekend, I have a small 5-months boy in house, I'm having difficulties finding strength and time to do non-work coding : o. I am working on it, thought, I have the technical side figured out and I'll open a MVP as soon as it's ready.

**Comment by Floppy** [comment]
Oh, I know that feeling well my friend. No rush or commitment at all :)

**Comment by lstrzebinczyk** [comment]
@Floppy So now that you merged my PR with implementation of these informations for kriegs and kommandos, my question is: what now?

There are 2 points I'd like to discuss:
1. There clearly are faction-specific rules and references. Instead of having faction-specific data, which then would be used in `Roster` to be rendered, there might be a faction-specific footer. A component, which could be imported in `Roster` and rendered on the bottom. It would include ploys, tac ops and whatever faction-specific information are there.
2. There are all the factions from Compendium, which also have ploys and tac ops that need to be migrated.

WDYT? Since I started this, I'm happy to help pushing both of those.

**Comment by Floppy** [comment]
So, the big thing I'm wondering is whether we should split out the data files into a separate repo - both to avoid copyright strikes somewhat and to share with other apps (including things like translations). I'm in early discussions with @genestealercult, creator of https://datacard.app about sharing translations, and I wonder if @rweyrauch may also be interested for their PrettyScribe tool.

I think I'd rather keep the faction stuff as data rather than components, both for that reason and for decoupling in general.

**Comment by lstrzebinczyk** [comment]
Alright then, when I can find some time, I'll open a PR with some additional faction-specific data, upholding current solution, you figure out the architecture: )

**Comment by Floppy** [comment]
Perfect, that sounds great 👍🏻

**Comment by rweyrauch** [comment]
Prettyscribe currently only pulls data from the Battlescribe roster files and does not use any of the catalog information directly.   

So far I have not have anyone ask for translations.  However, this may reflect more on the current BS users and less on a desire for non-english users.

**Comment by Floppy** [comment]
@rweyrauch same for this app and datacard, too. We're wondering about creating some shared files to maintain extra data that's not in the roster or catalogue. Sorry for randomly tagging you in with no context, but if you're interested, I'll keep you in the loop.

**Comment by lstrzebinczyk** [comment]
I'm taking on Brood Coven and Hive Fleet

**Comment by g3rg** [comment]
If no one has started, I'm happy to do the next few on the list (I own 3 of the 4 factions):
 Space Marine
 Grey Knight
 Imperial Guard
 Forge World

**Comment by Floppy** [comment]
I've created https://github.com/KTDataCollective/KT21_Factions, and may move data files over there when I get a couple of hours to code. I'll sort out access as and when I do.

**Comment by g3rg** [comment]
I've submitted another PR, for a few more. @Floppy would you rather I held off completing the data files until you've done a restructure? I'm happy to help with any reformatting after.

**Comment by Floppy** [comment]
No, it's fine, I'll just copy them all across in one go.

**Comment by Floppy** [comment]
BTW, I managed to finally get some time to check all this over, and it's great work, really, thank you both @lstrzebinczyk @g3rg :)

**Comment by lstrzebinczyk** [comment]
@g3rg Very nice! I'll try and add a batch today evening as well.

**Comment by g3rg** [comment]
I've got a friend who wants to get involved as well, who plays Necrons and Chaos Daemons, and I have the White Dwarf with the new Hunter Cadre in it, so we'll do those, and leave the others for you to choose from @lstrzebinczyk !

**Comment by Floppy** [comment]
Fantastic :)

**Comment by lstrzebinczyk** [comment]
@g3rg Please check whichever you'll be doing in the checklist, I'll take whichever are available.

**Comment by g3rg** [comment]
I'm not able to edit your post to check the boxes. The three factions we'll take care of in the next day or so are:

 Chaos Daemon
 Tomb World
 Hunter Cadre

**Comment by Floppy** [comment]
I ticked them 👍🏻

**Comment by Floppy** [comment]
We've not looked into it in detail yet, there's a ticket open for it though in #549.

**Comment by lstrzebinczyk** [comment]
I kinda think we should include all of the additional faction-specific data in our files and just add it on the bottom based on the faction on the roster. That will be much cleared as far as architecture goes.

**Comment by lstrzebinczyk** [comment]
Alright, I'm taking on "Cadre Marcenary" and "Greenskin"

**Comment by lstrzebinczyk** [comment]
Taking "Troupe" as well.

**Comment by g3rg** [comment]
I'll do the last 2 on the list, my friend should be submitting a PR for Daemons in the next day or so, he ran into some fun with what he thought was the Battlescribe file not matching the books... though then found out it was also Games Workshop themselves who were inconsistent.

Craftworld
Commorrite

**Comment by Floppy** [comment]
Looks like we're there folks, nice work! Just in time for Warpcoven in WD in a couple of weeks ;)
